internal/cmd/oohelperd/internal/nwcth/http.go

```diff
@@ -0,0 +1,110 @@
+package nwcth
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+
+	"github.com/ooni/probe-cli/v3/internal/iox"
+	"github.com/ooni/probe-cli/v3/internal/runtimex"
+)
+
+// HTTPConfig configures the HTTP check.
+type HTTPConfig struct {
+	Jar       *cookiejar.Jar
+	Headers   map[string][]string
+	Transport http.RoundTripper
+	URL       *url.URL
+}
+
+// HTTPDo performs the HTTP check.
```

Comment:
Can you also document what is returned and why? Something like `CtrlHTTPRequest is [...] and NextLocationInfo is [...]`. Make sure you mention when they can be `nil` (if they can). In general, it would be good to mention when a function returns something and when it returns `nil`. (Absent an optional type, it is quite useful to have a way to represent the absence via `nil`.)